proxy/proxy.go

```diff
@@ -470,6 +485,23 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, error) {
 	return response, nil
 }
 
+func (p *Proxy) checkBreaker(c *context) (func(bool), bool) {
+	if p.breakers == nil {
+		return nil, true
+	}
+
+	settings, _ := c.stateBag[circuitfilters.RouteSettingsKey].(circuit.BreakerSettings)
+	settings.Host = c.outgoingHost
```

Comment:
Can I set breakers for `<shunt>` or `<loopback>`? Is that desirable behaviour? Or should we check and early escape in those cases?